File: onboarding-portal/frontend/src/types/quickAssess.ts
export interface AssessmentFile {
  filename: string;
  status: 'pending' | 'processing' | 'completed' | 'failed';
  message?: string;
}

export type AssessmentStage =
  | 'ingestion'
  | 'parsing'
  | 'analysis'
  | 'report'
  | 'completed';

export interface AssessmentStatus {
  assessment_id: string;
  status: 'processing' | 'completed' | 'failed';
  stage: AssessmentStage;
  progress: number;
  estimated_seconds_remaining?: number;
  files: AssessmentFile[];
  error?: string;
  report_url?: string;
  report_json_url?: string;
  updated_at?: string;
}

export interface AssessmentResultSummary {
  cloud_readiness_score: number;
  key_findings: string[];
  infrastructure_counts: Record<string, number>;
  technology_stack: {
    languages: string[];
    frameworks: string[];
    cloud_services: string[];
    databases: string[];
    storage: string[];
  };
  risk_level: 'low' | 'medium' | 'high';
  report_url: string;
  report_json_url?: string;
  share_url?: string;
  recommendations?: {
    actions: {
      priority: 'High' | 'Medium' | 'Low';
      description: string;
    }[];
    timeline?: string;
  };
}

export interface AssessmentListItem {
  assessment_id: string;
  status: string;
  stage?: string;
  progress?: number;
  upload_time?: string;
  report_url?: string;
  cloud_readiness_score?: number;
}

export interface AssessmentListResponse {
  items: AssessmentListItem[];
  next_token?: string | null;
}

export interface UploadedAssessmentFileSummary {
  filename: string;
  size_bytes: number;
}

export interface QuickAssessUploadResponse {
  assessment_id: string;
  status: string;
  files: UploadedAssessmentFileSummary[];
}

export interface WorkflowLaunchResponse {
  assessment_id: string;
  workflow_id: string;
  status: string;
}

File: onboarding-portal/backend/app/schemas/quick_assess.py
"""Schemas for Quick Assess feature."""
from typing import Any, Dict, List, Optional
from pydantic import BaseModel, Field


class UploadedAssessmentFile(BaseModel):
    """Details about an uploaded assessment file."""

    filename: str = Field(..., description="Original file name")
    size_bytes: int = Field(..., description="Size of the uploaded file in bytes")


class QuickAssessUploadResponse(BaseModel):
    """Response returned after uploading assessment files."""

    assessment_id: str = Field(..., description="Generated assessment identifier")
    status: str = Field(default="uploaded", description="Current processing status")
    files: List[UploadedAssessmentFile]


class WorkflowLaunchResponse(BaseModel):
    """Response for workflow execution trigger."""

    assessment_id: str
    workflow_id: str
    status: str = Field(default="processing")


class AssessmentFileStatus(BaseModel):
    filename: str
    status: str
    message: Optional[str] = None


class AssessmentStatusResponse(BaseModel):
    assessment_id: str
    status: str
    stage: str
    progress: int = Field(ge=0, le=100)
    estimated_seconds_remaining: Optional[int] = None
    files: List[AssessmentFileStatus] = []
    error_messages: List[str] = []
    report_url: Optional[str] = None
    report_json_url: Optional[str] = None


class AssessmentResultResponse(BaseModel):
    assessment_id: str
    status: str
    report_url: Optional[str]
    report_json_url: Optional[str]
    results: Dict[str, Any]


class AssessmentListItem(BaseModel):
    assessment_id: str
    status: str
    stage: Optional[str] = None
    progress: Optional[int] = None
    upload_time: Optional[str] = None


class AssessmentListResponse(BaseModel):
    items: List[AssessmentListItem]
    next_token: Optional[str] = None


class ShareLinkResponse(BaseModel):
    assessment_id: str
    share_url: str

